### PR TITLE
Internalise the XXX_get_numbers() calls

### LINKED ISSUE #15530: Make the various XXX_get_number() calls private
These calls are effectively useless for applications.

They are useful internally so must be retained in some form.

### PR DESCRIPTION
**Pull request by paulidale**
Fixes #15530

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: doc/man3/OSSL_ENCODER.pod [hunk 1]
```diff
@@ -73,10 +71,7 @@ with the given I<encoder>.
 OSSL_ENCODER_is_a() checks if I<encoder> is an implementation of an
 algorithm that's identifiable with I<name>.
 
-OSSL_ENCODER_get_number() returns the internal dynamic number assigned to
-the given I<encoder>.
-
-OSSL_ENCODER_get_number() returns the name used to fetch the given I<encoder>.
+ossl_encoder_get0_name() returns the name used to fetch the given I<encoder>.
```
Comment (t8m) [inline doc/man3/OSSL_ENCODER.pod:74]: Uppercase OSSL_ENCODER_get0_name

File: doc/man3/OSSL_DECODER.pod [hunk 2]
```diff
@@ -73,10 +71,7 @@ with the given I<decoder>.
 OSSL_DECODER_is_a() checks if I<decoder> is an implementation
 of an algorithm that's identifiable with I<name>.
 
-OSSL_DECODER_get_number() returns the internal dynamic number assigned
-to the given I<decoder>.
-
-OSSL_DECODER_get_number() returns the name used to fetch the given I<decoder>.
+ossl_decoder_get0_name() returns the name used to fetch the given I<decoder>.
```
Comment (t8m) [inline doc/man3/OSSL_DECODER.pod:74]: Uppercase OSSL_DECODER_get0_name

**Review by t8m — CHANGES_REQUESTED** [review]

File: doc/man3/OSSL_ENCODER.pod (hunk 1 above)
Comment (paulidale) [reply doc/man3/OSSL_ENCODER.pod]: Doh!

Fixed.

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Merged to master, thanks for the reviews.

**Pull request closed without merging**